tooling/nargo/src/foreign_calls/print.rs

```diff
@@ -8,12 +8,24 @@ use noirc_printable_type::{PrintableType, PrintableValueDisplay};
 
 use super::{ForeignCall, ForeignCallError, ForeignCallExecutor};
 
-#[derive(Debug, Default)]
+#[derive(Default)]
 pub enum PrintOutput<'a> {
     #[default]
     None,
     Stdout,
     String(&'a mut String),
+    PrintCallback(Box<dyn Fn(String) + 'a>),
```

Comment:
I've been warning to remove this enum and just pass in a writer for any output to be sent to for some time. If we're adding a callback here I think we've reached the point of where we should address this.